[Az.StorageSync] Remove alias "RegisteredServer" for InputObject parameter of Set-AzStorageSyncServerEndpoint

anpint opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

The purpose of this PR is to make a breaking change that has been documented for our module Az.StorageSync for some time now: https://github.com/Azure/azure-powershell/blob/90ab7fc6075e8f62947774160a0eaf239cbc324b/documentation/breaking-changes/upcoming-breaking-changes/Az.StorageSync-UpcomingBreakingChanges.md?plain=1#L19

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

msJinLei commented:
please suppress the breaking change error
```
2023-05-08T21:39:40.5821269Z   D:\a\_work\1\s\artifacts//StaticAnalysisResults\BreakingChangeIssues.csv Errors
2023-05-08T21:39:40.5821854Z   "Module","ClassName","Target","Severity","ProblemId","Description","Remediation"
2023-05-08T21:39:40.5822537Z   "Az.StorageSync","Microsoft.Azure.Commands.StorageSync.Cmdlets.SetServerEndpointCommand","Set-AzStorageSyncServerEndpoint","0","2010","The cmdlet 'Set-AzStorageSyncServerEndpoint' no longer supports the alias 'RegisteredServer' for parameter 'InputObject'.","Add the alias 'RegisteredServer' back to parameter 'InputObject'."
``

```
suppress the breaking change errors in "tools/StaticAnalysis/Exceptions/${ModuleName}/BreakingChangeIssues.csv"
"AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
"Microsoft.Azure.PowerShell.Cmdlets.Network.dll","Microsoft.Azure.Commands.Network.NewAzFirewallPolicyNatRuleCommand","New-AzFirewallPolicyNatRule","0","1050","The parameter set '__AllParameterSets' for cmdlet 'New-AzFirewallPolicyNatRule' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'New-AzFirewallPolicyNatRule'."

remove the unnecessary breaking change notifications
```

msJinLei commented:
/azp run azure-powershell - powershell-core

msJinLei commented:
Please provide the migration guide of StorageSync for this major release? 

Please fill in the following table 
|cmdlet|breaking change|old usage|new usage| 
|-|-|-|-| 

Thanks

msJinLei commented:
@anpint 

> **Please note the code freeze day is next Monday. Please finish your PR in this weekend (take review and return revision time into consideration). For this release is very important and complicate, we are unprobable to merge PR after code freeze. Thank you for your understanding and cooperation.**

msJinLei commented on documentation/breaking-changes/upcoming-breaking-changes/Az.StorageSync-UpcomingBreakingChanges.md:
```diff
@@ -5,17 +5,6 @@
 
 
 
-### **Invoke-AzStorageSyncFileRecall**
```
Please revert the change to this file

anpint replied:
Reverted

msJinLei commented:
/azp run azure-powershell - security-tools

anpint commented:
> Please provide the migration guide of StorageSync for this major release.
> 
> Please fill in the following table
> 
> cmdlet	breaking change	old usage	new usage
> Thanks

| cmdlet | breaking change | old useage | new usage |
|---------|--------------------|-------------|-------------|
| Set-AzStorageSyncServerEndpoint | Removed "RegisteredServer" alias for parameter "InputObject"  | Set-AzStorageSyncServerEndpoint -RegisteredServer $inputObject  | Set-AzStorageSyncServerEndpoint -ServerEndpoint $inputObject |

anpint commented:
@msJinLei could you please confirm if the window closed yesterday? If so I'll abandon this PR since we didn't make the window.

anpint commented:
> > @msJinLei could you please confirm if the window closed yesterday? If so I'll abandon this PR since we didn't make the window.
> 
> @anpint You PR don't have too much to change. Just to revert the change to [Az.StorageSync-UpcomingBreakingChanges.md](https://github.com/Azure/azure-powershell/pull/21730/files#diff-56c6d57a6c3c1e85933e8f0c8b9e9b2945f7d72258d89955c7bfe09bfc98f680) and suppress the breaking change. If you can finish it by today. We can include your PR into the RC2.

I believe I've addressed all of your comments at this point. Let me know if there are further changes needed.

msJinLei commented:
/azp run azure-powershell - security-tools

anpint commented:
@msJinLei Thanks for helping to get this PR merged! And sorry for the inconvenience of missing the deadline on Monday.

msJinLei commented:
> @msJinLei Thanks for helping to get this PR merged! And sorry for the inconvenience of missing the deadline on Monday.

@anpint  Thank you for understanding
